### PR TITLE
Sarah query fixes

### LINKED ISSUE #36: Revise query get_parent_class code
Matthew tidied this up and commented out some code for Sarah to look at.

### LINKED ISSUE #37: Look at query isV2BaseAttribute function
This probably needs to be more generic

### LINKED ISSUE #38: Look at query get_max_length
TODO will this throw an exception if elements other
    than the first one do *not* have the attribute?
    Is that possible?
    "

### PR DESCRIPTION
**Pull request by skeating**
## Description

Fixes for the query file 

## Motivation and Context

fixes #36 
fixes #37 
fixes #38

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Refactor/documentation

## Checklist:

- [x] I have updated all documentation necessary.
- [x] I have checked spelling in (new) comments.

### CODE REVIEW

File: generator/util/query.py [hunk 1]
```diff
@@ -276,14 +276,13 @@ def get_inline_parents(class_object):
 
 def get_parent_class(class_object):
     """
-    NB Open Github Issue: Revise query get_parent_class code #36
-
     Return the parent class of this class.
-    if it is not given we assume it is a child of a plugin
-    and get the base of the plugin.
+    If a class does not explicitly declare its parent then it's parent may
```
Comment (mattagape) [inline generator/util/query.py]: typo, it's should be its

File: generator/util/query.py [hunk 2]
```diff
@@ -276,14 +276,13 @@ def get_inline_parents(class_object):
...
+    be the outside this package i.e. a plugin object. In this case it will be
```
Comment (mattagape) [inline generator/util/query.py]: typo, unnecessary 'the'

File: generator/util/query.py [hunk 3]
```diff
@@ -300,55 +299,72 @@ def get_parent_class(class_object):
...
           <attributes>
-             <attribute name="unit" required="false" type="element" element="Unit" abstract="false"/>
+             <attribute name="unit" required="false" type="element"
+                        element="Unit" abstract="false"/>
           </attributes>
        </element>
-    """
-    parent = ''
+
+    e.g. in the following XML, the 'Def' element is reerenced by the
```
Comment (mattagape) [inline generator/util/query.py]: typo reerenced -> referenced

File: generator/util/query.py [hunk 4]
```diff
@@ -300,55 +299,72 @@ def get_parent_class(class_object):
...
+
+    would return
+
+    .. code-block:: default
+
+       'Model'
+
+    .. code-block:: xml
+
+      <elements>
```
Comment (mattagape) [inline generator/util/query.py]: I think this XML block needs to be indented another space or two, or it won't be nicely formatted in Sphinx.

Comment (mattagape) [reply generator/util/query.py]: It must be indented at least as far as the 'c' at the start of 'code-block'

**Review by mattagape — CHANGES_REQUESTED** [review]

File: generator/util/query.py (hunk 2 above)
Comment (skeating) [reply generator/util/query.py]: ???

Comment (mattagape) [reply generator/util/query.py]: @skeating  'be the outside this package' - no need for 'the' here

**Review by mattagape — APPROVED** [review]

**Pull request merged**